src/frequenz/sdk/actor/_actor.py

```diff
@@ -54,6 +58,20 @@ def start(self) -> None:
     async def _run(self) -> None:
         """Run this actor's logic."""
 
+    async def _delay_if_restart(self, iteration: int) -> None:
+        """Delay the restart of this actor's n'th iteration.
+
+        Args:
+            iteration: The current iteration of the restart.
+        """
+        # NB: I think it makes sense (in the future) to think about deminishing returns
+        # the longer the actor has been running.
+        # Not just for the restart-delay but actually for the n_restarts counter as well.
+        if iteration > 0:
+            delay = Actor.RESTART_DELAY.total_seconds()
```

Comment:
I think you should say `self.RESTART_DELAY`, because then, actor implementations can override the interval if they want.

```suggestion
            delay = self.RESTART_DELAY.total_seconds()
```

Reply:
or maybe it is better to hide it and keep it internal for now? 🤔

Reply:
done